src/arrayMethodJoin.js

```diff
@@ -4,8 +4,24 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    if (this.length === 1) {
+      return `${this[0]}`;
+    }
+
+    let result = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (this[i] !== undefined && this[i] !== null) {
+        result += `${this[i]}`;
```

Comment:
```suggestion
        result += this[i];
```
There is no need to use template literals. The value will be converted into string anyway